### PR TITLE
Add support for CheckedStatement for smart break line

### PR DESCRIPTION
**Pull request by Cosifne**
Fix https://github.com/dotnet/roslyn/issues/60250

### CODE REVIEW

File: src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler_Helpers.cs [hunk 1]
```diff
@@ -642,6 +643,11 @@ private static bool ShouldAddBraceForFinallyClause(FinallyClauseSyntax finallyCl
                && finallyClauseNode.Block.OpenBraceToken.IsMissing
                && !finallyClauseNode.Block.Span.Contains(caretPosition);
 
+        private static bool ShouldAddBraceForCheckedStatement(CheckedStatementSyntax checkedStatementNode, int caretPosition)
+            => checkedStatementNode.Block.OpenBraceToken.IsMissing
+                && checkedStatementNode.Block.Statements.IsEmpty()
+                && !checkedStatementNode.Block.Span.Contains(caretPosition);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler_Helpers.cs:649]: why the statements.isempty check?  i don't see that on any of hte others.

File: src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs [hunk 2]
```diff
@@ -619,6 +619,7 @@ private static int GetBraceInsertionPosition(SyntaxNode node)
                 TryStatementSyntax tryStatementNode => tryStatementNode.TryKeyword.Span.End,
                 CatchClauseSyntax catchClauseNode => catchClauseNode.Block.SpanStart,
                 FinallyClauseSyntax finallyClauseNode => finallyClauseNode.Block.SpanStart,
+                CheckedStatementSyntax checkedStatementNode => checkedStatementNode.Keyword.Span.End,
```
Comment (dibarbet) [inline src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs:622]: does this also include unchecked?

File: src/EditorFeatures/CSharpTest/AutomaticCompletion/AutomaticLineEnderTests.cs [hunk 3]
```diff
@@ -3132,6 +3132,30 @@ public void Bar2()
 }");
         }
 
+        [WpfFact]
+        public void TestCheckedStatement()
```
Comment (dibarbet) [inline src/EditorFeatures/CSharpTest/AutomaticCompletion/AutomaticLineEnderTests.cs:3136]: a test that verifies it also works in `unchecked` and one to verify that nothing happens for `var m = checked(1+1)` would be good to have I think

File: src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs (hunk 2 above)
Comment (Cosifne) [reply src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs:622]: Yes, I think checked statement includes both, with the different keyword token

File: src/EditorFeatures/CSharpTest/AutomaticCompletion/AutomaticLineEnderTests.cs (hunk 3 above)
Comment (Cosifne) [reply src/EditorFeatures/CSharpTest/AutomaticCompletion/AutomaticLineEnderTests.cs:3136]: Sure

File: src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler_Helpers.cs (hunk 1 above)
Comment (Cosifne) [reply src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler_Helpers.cs:649]: : ) I will remove this one then.

File: src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs (hunk 2 above)
Comment (Cosifne) [reply src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs:622]: Tests added

**Review by dibarbet — APPROVED** [review]

**Pull request merged**